core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -117,4 +120,42 @@ private static void deleteFiles(FileIO io, Set<ManifestFile> allManifests) {
           }
         });
   }
+
+  /**
+   * Load FileIO, default to HadoopFileIO
+   * @param tableMetadata table metadata
+   * @param conf Hadoop configuration
+   * @return FileIO class
+   * @throws IllegalArgumentException if class path not found or
+   *  right constructor not found or
+   *  the loaded class cannot be casted to the given interface type
+   */
+  public static FileIO loadFileIO(TableMetadata tableMetadata, Configuration conf) {
+    if (tableMetadata != null) {
+      Map<String, String> properties = tableMetadata.properties();
```

Comment:
Looks like we only need `properties` here. I think that we should probably pass a map of properties to this method instead. That way changes to table metadata won't need to affect this class.